src/dev/flang/tools/fzjava/ForClass.java

```diff
@@ -474,31 +474,45 @@ void processMethod(Method me,
         data_dynamic.append("\n" +
                             "  # call Java instance method '" + me + "':\n" +
                             "  #\n" +
-                            "  public " + fn + fp + " " + fr + " =>\n" +
-                            "    " + ("fuzion.java.call_virtual (" + fr + ") " +
+                            "  public " + fn + fp + " " + complexResultType(me, fr) + " =>\n" +
+                            "    " + ("match fuzion.java.call_virtual (" + fr + ") " +
                                       fuzionString(_class.getName()) + " " +
                                       fuzionString(jn) + " " +
                                       fuzionString(js) + " " +
                                       fcn + ".this "+
-                                  parametersArray(outer + "." + fn, pa) + "\n")
+                                  parametersArray(outer + "." + fn, pa) + "\n") +
+                            "      " + "e error => " + (hasCheckedExceptions(me) ? "e" : "panic e.msg") + "\n" +
+                            "      " + "r " + fr + " => r\n"
                             );
       }
     else
       {
         data_static.append("\n" +
                             "  # call Java static method '" + me + "':\n" +
                             "  #\n" +
-                            "  public " + fn + fp + " " + fr + " =>\n" +
-                            "    " + ("fuzion.java.call_static (" + fr + ") " +
+                            "  public " + fn + fp + " " + complexResultType(me, fr) + " =>\n" +
+                            "    " + ("match fuzion.java.call_static (" + fr + ") " +
                                       fuzionString(me.getDeclaringClass().getName()) + " " +
                                       fuzionString(jn) + " " +
                                       fuzionString(js) + " " +
-                                  parametersArray(outer + STATIC_SUFFIX + "." + fn, pa) + "\n")
+                                  parametersArray(outer + STATIC_SUFFIX + "." + fn, pa) + "\n") +
+                            "      " + "e error => " + (hasCheckedExceptions(me) ? "e" : "panic e.msg") + "\n" +
+                            "      " + "r " + fr + " => r\n"
                             );
       }
   }
 
 
+  /**
+   * If excecutable throws any checked exception
+   * the result type `fr` is wrapped in an outcome.
+   */
+  private String complexResultType(Executable exc, String fr)
+  {
+    return hasCheckedExceptions(exc) ? "outcome("+fr+")" : fr;
```

Comment:
needs parens currently, error without e.g.:
```
build/modules/java.base/Java/java/lang/Class.fz:168:29: error 10: Wrong number of type parameters
  public getDeclaredClasses outcome fuzion.java.Array (Java.java.lang.Class) =>
----------------------------^^^^^^^
Wrong number of actual type parameters in type:
Type: outcome fuzion.java.Array Java.java.lang.Class
expected one generic argument for 'T'
found 2: '**error**, Java.java.lang.Class'.

```

will create issue though

Reply:
Does not work either because then i get stuff like:

build/modules/java.base/Java/java/net/MulticastSocket_static.fz:35:10: error 10: Duplicate feature declaration
  public new(arg0 Java.java.net.SocketAddress) =>
---------^^^
Feature that was declared repeatedly: 'Java.java.net.MulticastSocket_static.new'
originally declared at build/modules/java.base/Java/java/net/MulticastSocket_static.fz:28:10:
  public new (outcome Java.java.net.MulticastSocket) =>
---------^^^